doc/shared_array_buffer.md

```diff
@@ -0,0 +1,65 @@
+# SharedArrayBuffer
```

Comment:
You should add this as link to the `README` under section

API Documentation -> Data Types